src/lib/route_rule.rs

```diff
@@ -0,0 +1,206 @@
+use crate::netlink::parse_as_ipv4;
+use crate::netlink::parse_as_ipv6;
+use crate::route::AddressFamily;
+use crate::route::RouteProtocol;
+use crate::NisporError;
+use futures::stream::TryStreamExt;
+use netlink_packet_route::rtnl::rule::nlas::Nla;
+use netlink_packet_route::RuleMessage;
+use rtnetlink::new_connection;
+use rtnetlink::IpVersion;
+use serde_derive::{Deserialize, Serialize};
+use tokio::runtime::Runtime;
+
+const FR_ACT_TO_TBL: u8 = 1;
+const FR_ACT_GOTO: u8 = 2;
+const FR_ACT_NOP: u8 = 4;
+const FR_ACT_RES3: u8 = 8;
+const FR_ACT_RES4: u8 = 16;
+const FR_ACT_BLACKHOLE: u8 = 32;
+const FR_ACT_UNREACHABLE: u8 = 64;
+const FR_ACT_PROHIBIT: u8 = 128;
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub enum RuleAction {
+    /* Pass to fixed table */
+    Table,
+    /* Jump to another rule */
+    Goto,
+    /* No operation */
+    Nop,
+    Res3,
+    Res4,
+    /* Drop without notification */
+    Blackhole,
+    /* Drop with ENETUNREACH */
+    Unreachable,
+    /* Drop with EACCES */
+    Prohibit,
+    Other(u8),
+    Unknown,
+}
+
+impl From<u8> for RuleAction {
+    fn from(d: u8) -> Self {
+        match d {
+            FR_ACT_TO_TBL => Self::Table,
+            FR_ACT_GOTO => Self::Goto,
+            FR_ACT_NOP => Self::Nop,
+            FR_ACT_RES3 => Self::Res3,
+            FR_ACT_RES4 => Self::Res4,
+            FR_ACT_BLACKHOLE => Self::Blackhole,
+            FR_ACT_UNREACHABLE => Self::Unreachable,
+            FR_ACT_PROHIBIT => Self::Prohibit,
+            _ => Self::Other(d),
+        }
+    }
+}
+
+impl Default for RuleAction {
+    fn default() -> Self {
+        Self::Unknown
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone, Default)]
+pub struct RouteRule {
+    pub action: RuleAction,
+    pub address_family: AddressFamily,
+    pub flags: u32,
+    pub table: u32,
```

Comment:
We should hide `table` when it is `RT_TABLE_UNSPEC`

Reply:
Done. Thanks!